rpcserver.go

```diff
@@ -531,17 +534,33 @@ func newRPCServer(s *server, macService *macaroons.Service,
 
 	// If macaroons aren't disabled (a non-nil service), then we'll set up
 	// our set of interceptors which will allow us handle the macaroon
-	// authentication in a single location .
-	if macService != nil {
-		unaryInterceptor := grpc.UnaryInterceptor(
+	// authentication in a single location. If monitoring is enabled, we'll
+	// also set up interceptors for the Prometheus exporter. If both are
+	// enabled, we'll chain the interceptors together.
+	var unaryInterceptor grpc.UnaryServerInterceptor
+	var streamInterceptor grpc.StreamServerInterceptor
+	switch {
+	case macService != nil && monitoring.Enabled:
+		unaryInterceptor = grpc_middleware.ChainUnaryServer(
 			macService.UnaryServerInterceptor(permissions),
+			grpc_prometheus.UnaryServerInterceptor,
 		)
-		streamInterceptor := grpc.StreamInterceptor(
+		streamInterceptor = grpc_middleware.ChainStreamServer(
 			macService.StreamServerInterceptor(permissions),
+			grpc_prometheus.StreamServerInterceptor,
 		)
+	case macService != nil:
+		unaryInterceptor = macService.UnaryServerInterceptor(permissions)
+		streamInterceptor = macService.StreamServerInterceptor(permissions)
+	case monitoring.Enabled:
```

Comment:
Rather than check this manually here independent of the build tag, I think this can be melded in with the current `monitoring.Start` method.

Reply:
Hm, server options need to be set before any services can register with the grpc server. Lmk if my solution works.